packages/rum-recorder/src/boot/recorder.ts

```diff
@@ -0,0 +1,67 @@
+import { Configuration } from '@datadog/browser-core'
+import { LifeCycle, ParentContexts } from '@datadog/browser-rum-core'
+
+import { DeflateSegmentWriter } from '../domain/deflateSegmentWriter'
+import { createDeflateWorker } from '../domain/deflateWorker'
+import { record } from '../domain/rrweb'
+import { startSegmentCollection } from '../domain/segmentCollection'
+import { trackSegmentRenewal } from '../domain/trackSegmentRenewal'
+import { send, SEND_BEACON_BYTE_LENGTH_LIMIT } from '../transport/send'
+
+export function startRecording(
+  lifeCycle: LifeCycle,
+  applicationId: string,
+  configuration: Configuration,
+  parentContexts: ParentContexts
+) {
+  const worker = createDeflateWorker()
+
+  const writer = new DeflateSegmentWriter(
+    worker,
+    (size) => {
+      if (size > SEND_BEACON_BYTE_LENGTH_LIMIT) {
+        renewSegment('max_size')
+      }
+    },
+    (data, meta) => {
+      send(configuration.sessionReplayEndpoint, data, meta)
+    }
+  )
+
+  const { addRecord, renewSegment } = startSegmentCollection(
+    () => getSegmentContext(applicationId, parentContexts),
+    writer
+  )
+
+  const { stop: stopSegmentRenewal } = trackSegmentRenewal(lifeCycle, renewSegment)
+
+  const stopRecording = record({
+    emit: addRecord,
+  })!
```

Comment:
IMO, some of the setup made here would also make sense in segment collection.

What about:

- in recorder, only wire segment collection with record through addRecord method
- segment collection is responsible to create worker and writer, and to define segment context
- extract segment file from segment collection

it would allow to:
- have all conditions to renew the segment declared at the same place
- remove trackSegmentRenewal
- separate segment lifecycle from segment content